### PR TITLE
Update pane initialization and display logic

### PR DESCRIPTION
**Pull request by amcgee**
Allow initially-hidden non-collapsible panes, sanitize togglePane 'show' parameter, and publish a subscribable event when toggling a pane. This allows for better declarative configuration without needing to utilize CSS properties or javascript callbacks.

Note : setting "open: false" still causes the sidebar to flash before disappearing. This is because the initial CSS is still set to display 'block'. To fix this, the pane initialization function would need to modify the CSS of the sidebar before it is rendered.

### CODE REVIEW

**Comment by tmcgee** [comment]
@amcgee Please merge the new changes from the cmv-app develop branch and then I will merge in your enhancement. Thanks!

**Comment by amcgee** [comment]
Updated, thanks!

**Comment by roemhildtg** [comment]
This seems to be causing a problem. When a pane is toggled via a collapse button it no longer works. It looks like line 182 will send the key, and the domEvent as a second argument. 

```
on(this.collapseButtons[key], 'click', lang.hitch(this, 'togglePane', key));
```

The new logic doesn't check for this, and throws an error.

File: viewer/js/viewer/_LayoutMixin.js [hunk 1]
```diff
@@ -213,20 +213,46 @@ define([
...
+                    // Set (CSS Display Property)
+                    newDisp = show;
+                } else if (typeof(show) === 'boolean') {
+                    // Set (boolean)
+                    newDisp = (show) ? 'block' : 'none';
+                } else if (show === undefined) {
+                    // Toggle
+                    newDisp = (oldDisp === 'none') ? 'block' : 'none';
+                } else {
+                    this.handleError({
```
Comment (roemhildtg) [inline viewer/js/viewer/_LayoutMixin.js]: Need to replace this error message with this: 
```
newDisp =(domStyle.get(domNode, 'display') === 'none') ? 'block' : 'none';
```

Comment (amcgee) [inline viewer/js/viewer/_LayoutMixin.js]: Good catch, thanks @roemhildtg!  While accepting any type of 'show' argument and toggling if it isn't a display CSS string or boolean would work, another fix is to update the Toggle handler to call togglePane with three explicit arguments so we get togglePane(key, null, false, <domEvent>).  That keeps the togglePane API as strict as possible.  I will open a separate pull request for discussion about that approach.

**Pull request merged**